azure-pipelines.yml

```diff
@@ -68,3 +68,13 @@ jobs:
   parameters:
     name: Windows
     vmImage: windows-2019
+
+- job: check_formatting
+  pool:
+    vmImage: 'ubuntu-latest'
+  steps:
+  - script: |
+      set -e
+      sudo apt install -y clang-format
+      clang-format -n -Werror ffi/*.cpp ffi/*.h
+    displayName: Check C++ formatting
```

Comment:
Does `clang-format` produce the same formatting invariant of the `clang-format` version?

Also, I think it'd perhaps good to document that this is the formatting standard in the developer docs and issue some instructions about how to obtain and use it. I wonder if it would help to get the `clang-format` binary from the `clang-tools` conda package so as to provide a cross platform method of obtaining this tool too, it might make it easier in terms of development setup instructions?

Finally, is this adding a whole new build job to check formatting? Could this check be made as part of one of the existing build just by just adding an env var and the build script checking for that/adding it as a scripted step to an existing job? xref flake8 handling here: https://github.com/numba/numba/blob/61ec1fd0f69aeadece218dccf4c39ebc5c7dfbc4/buildscripts/azure/azure-linux-macos.yml#L30-L35

Reply:
Per private discussion, switching to the `clang-format-13` package from `conda-forge`.